### PR TITLE
SelectPanel2: Export props + pass props

### PR DESCRIPTION
**Pull request by siddharthkp**
- Closes https://github.com/github/primer/issues/3024

### Rollout strategy

- [x] Patch release

### Merge checklist

- [dont have tests yet!] Added/updated tests
- [x] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- NA] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

**Review by broccolinisoup — APPROVED** [review]

**Review by joshblack — APPROVED** [review]

**Comment by ayy-bc** [comment]
@siddharthkp we would also need the `SelectPanelSecondaryActionProps` to be exported to use it with the item picker for cases where we need secondary actions (related: https://github.com/github/collaboration-workflows-flex/issues/423, https://github.com/github/github/pull/310856/files). 

Do you want me to create another issue to track that or can that also go as a part of this PR?

File: src/drafts/SelectPanel2/SelectPanel.tsx [hunk 1]
```diff
@@ -250,6 +250,7 @@ const Panel: React.FC<SelectPanelProps> = ({
             '100%': {transform: 'scale(1, 1)'},
           },
         }}
+        {...props}
```
Comment (JelloBagel) [inline src/drafts/SelectPanel2/SelectPanel.tsx:253]: @siddharthkp Does the typescript on `props` also allow `StyledOverlay` props to be passed through? 

For context, I'm asking case where the prop `maxHeight` is needed to fix a styling issue with the overlay height: https://github.com/github/collaboration-workflows-flex/issues/449
```
<SelectPanel
      height="fit-content"
      maxHeight="large"
      ...
/>
```

**Comment by siddharthkp** [comment]
@ayy-bc Sure thing, I can add them to this PR

File: src/drafts/SelectPanel2/SelectPanel.tsx (hunk 1 above)
Comment (siddharthkp) [reply src/drafts/SelectPanel2/SelectPanel.tsx:253]: >  Does the typescript on props also allow StyledOverlay props to be passed through?

Not yet!

> I'm asking case where the prop maxHeight is needed to fix a styling issue with the overlay height: https://github.com/github/collaboration-workflows-flex/issues/449

That's an interesting issue, not sure if maxHeight is the solution. Going to reply on the issue first itself

**Pull request merged**